src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -404,6 +422,42 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
       data => {
         SELF.updateLeaderboardResults(data['results'], SELF);
         SELF.startLeaderboard(SELF.selectedPhaseSplit['id']);
+        SELF.sortLeaderboardTextOption = SELF.showLeaderboardByLatest ? 'Sort by best' : 'Sort by latest';
+      },
+      err => {
+        SELF.globalService.handleApiError(err);
+      },
+      () => {}
+    );
+  }
+
+  showLeaderboardByLatestOrBest() {
+    const API_PATH = this.endpointsService.particularChallengePhaseSplitUrl(this.selectedPhaseSplit['id']);
+    const SELF = this;
+    let BODY;
+
+    SELF.showLeaderboardByLatest = !SELF.showLeaderboardByLatest;
+
+    if (this.phaseSplits[this.selectedPhaseSplit['id']].hasOwnProperty('show_leaderboard_by_latest_submission')) {
```

Comment:
Same for here as I've explained on the above comment, you can remove these if conditions and can directly assign the `!`(not) of variable like.

`BODY = JSON.stringify({ 'show_leaderboard_by_latest_submission': !SELF.showLeaderboardByLatest });`

Reply:
Changed!

Reply:
@drepram You haven't made this change here. Remove this if and else case here just change it to `BODY = JSON.stringify({ 'show_leaderboard_by_latest_submission': !SELF.showLeaderboardByLatest });`